frontend/amundsen_application/static/js/components/PopularResources/index.tsx

```diff
@@ -0,0 +1,138 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+
+import { PopularResource, ResourceDict, ResourceType } from 'interfaces';
+
+import InfoButton from 'components/InfoButton';
+import PaginatedResourceList from 'components/ResourceList/PaginatedResourceList';
+import TabsComponent, { TabInfo } from 'components/TabsComponent';
+import ShimmeringResourceLoader from 'components/ShimmeringResourceLoader';
+import {
+  getDisplayNameByResource,
+  indexDashboardsEnabled,
+} from 'config/config-utils';
+
+import popularResources, {
+  getPopularResources,
+} from 'ducks/popularTables/reducer';
+import { GetPopularResourcesRequest } from 'ducks/popularTables/types';
+import { GlobalState } from 'ducks/rootReducer';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import {
+  POPULAR_RESOURCES_LABEL,
+  POPULAR_RESOURCES_INFO_TEXT,
+  POPULAR_RESOURCES_SOURCE_NAME,
+  POPULAR_RESOURCES_PER_PAGE,
+  EMPTY_POPULAR_RESOURCES_MESSAGE,
+} from './constants';
+
+import './styles.scss';
+
+export interface StateFromProps {
+  popularResources: ResourceDict<PopularResource[]>;
+  isLoaded: boolean;
+}
+
+export interface DispatchFromProps {
+  getPopularResources: () => GetPopularResourcesRequest;
+}
+
+export type PopularResourcesProps = StateFromProps & DispatchFromProps;
+
+export class PopularResources extends React.Component<PopularResourcesProps> {
+  componentDidMount() {
+    this.props.getPopularResources();
+  }
+
+  generateTabContent = (resource: ResourceType): JSX.Element | undefined => {
+    const popularResources = this.props.popularResources[resource];
+
+    if (!popularResources) {
+      return undefined;
+    }
+
+    return (
+      <PaginatedResourceList
+        allItems={popularResources}
+        emptyText={EMPTY_POPULAR_RESOURCES_MESSAGE}
+        itemsPerPage={POPULAR_RESOURCES_PER_PAGE}
+        source={POPULAR_RESOURCES_SOURCE_NAME}
+      />
+    );
+  };
+
+  generateTabKey = (resource: ResourceType) => `popularresourcetab:${resource}`;
+
+  generateTabTitle = (resource: ResourceType): string => {
```

Comment:
I would pass here the popularResources prop and move this function out of the class